src/overlays/actors/ovl_En_Ru/z_en_ru.c

```diff
@@ -76,34 +86,269 @@ static DamageTable D_80A39488 = {
+        Actor_PlaySfxAtPos(&this->actor, sfxId);
+    }
+
+    this->isRightFootGrounded = isFootGrounded = SubS_IsFloorAbove(play, &this->rightFootPos, -6.0f);
+    if (this->isRightFootGrounded && !rightWasGrounded && isFootGrounded) {
+        Actor_PlaySfxAtPos(&this->actor, sfxId);
+    }
+
+    return false;
+}
+
+// This function is only ever called with eyeStateMax == 3
+void EnRu_UpdateEyes(EnRu* this, s32 eyeStateMax) {
+    if (DECR(this->blinkTimer) == 0) {
+        this->eyeState++;
+        if (this->eyeState >= eyeStateMax) {
+            this->eyeState = 0;
+            this->blinkTimer = Rand_S16Offset(30, 30);
+        }
+    }
+}
+
+void EnRu_UpdateCollider(EnRu* this, PlayState* play) {
+    this->collider.dim.pos.x = this->actor.world.pos.x;
+    this->collider.dim.pos.y = this->actor.world.pos.y;
+    this->collider.dim.pos.z = this->actor.world.pos.z;
+    CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+}
+
+void EnRu_UpdateModel(EnRu* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    SkelAnime_Update(&this->skelAnime);
+
+    // Head and Torso tracking to Player if in front of Ru.
+    if (SubS_AngleDiffLessEqual(this->actor.shape.rot.y, 0x2710, this->actor.yawTowardsPlayer)) {
+        Vec3f playerPos;
+
+        playerPos.x = player->actor.world.pos.x;
+        playerPos.y = player->bodyPartsPos[7].y + 3.0f;
+        playerPos.z = player->actor.world.pos.z;
+        SubS_TrackPoint(&playerPos, &this->actor.focus.pos, &this->actor.shape.rot, &this->trackTarget, &this->headRot,
+                        &this->torsoRot, &sTrackOptions);
+
+    } else { // smooth her back to facing forward
+        Math_SmoothStepToS(&this->trackTarget.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->trackTarget.y, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->headRot.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->headRot.y, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->torsoRot.x, 0, 4, 0x3E8, 1);
+        Math_SmoothStepToS(&this->torsoRot.y, 0, 4, 0x3E8, 1);
+    }
+
+    EnRu_UpdateEyes(this, 3);
+    EnRu_PlayWalkingSound(this, play);
+    SubS_FillLimbRotTables(play, this->limbRotTableY, this->limbRotTableZ, RU2_LIMB_MAX);
+}
+
+void EnRu_DoNothing(EnRu* this, PlayState* play) {
+}
+
+void EnRu_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnRu* this = THIS;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gAdultRutoSkel, NULL, this->jointTable, this->morphTable, RU2_LIMB_MAX);
+    EnRu_ChangeAnim(&this->skelAnime, 0);
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+
+    // path is prepared but unused after this
```

Comment:
```suggestion
```
Redundant now it's mentioned in the struct (and pretty clear if you just search for the variable name, this is not a large actor.